client/smarthomeconnect.cpp

```diff
@@ -0,0 +1,276 @@
+#include "smarthomeconnect.h"
+
+
+namespace smart_home_client
+{
+
+
+void SmartHomeConnect::StartConnection()
+{
+    std::unique_ptr<int, SocketDeleter> clientSocket(new int(socket(AF_INET, SOCK_STREAM, 0)));
+
+    if (*clientSocket == -1)
+    {
+        std::cerr << "Failed to create socket" << std::endl;
+        return;
+    }
+    
+	sockaddr_in serverAddress; 
+	serverAddress.sin_family = AF_INET; 
+	serverAddress.sin_port = htons(8080); 
+	serverAddress.sin_addr.s_addr = INADDR_ANY; 
+
+	if(connect(*clientSocket, (struct sockaddr*)&serverAddress, sizeof(serverAddress)) == -1)
+    {
+        std::cerr << "Failed to bind socket" << std::endl;
+        return;
+    } 
+
+    while (true)
+    {
+        std::string reqStr = RequestString();
+        if (reqStr == "X")
+        {
+            break;
+        }
+	    sendDataToServer(reqStr.c_str(), clientSocket.get());
+    }
+
+	const char* message = ""; 
+	int bytesSent = send(*clientSocket, message, strlen(message), 0); 
+    if (bytesSent == -1) {
+        std::cerr << "Failed to send data to server" << std::endl;
+        return;
+    }
+}
+
+void SmartHomeConnect::sendDataToServer(const char* message, int* clientSocket)
+{
+    int bytesSent = send(*clientSocket, message, strlen(message), 0); 
+    if (bytesSent == -1) {
+        std::cerr << "Failed to send data to server" << std::endl;
+        return;
+    }
+
+    char buffer[1024] = {0};
+    int bytesReceived = recv(*clientSocket, buffer, sizeof(buffer), 0);
+    if (bytesReceived == -1) {
+        std::cerr << "Failed to receive data from server" << std::endl;
+        return;
+    }
+
+    std::cout << "\nReceived from server:\n" << buffer << std::endl;
+}
+
+std::string SmartHomeConnect::RequestString()
+{
+    std::cout<<"Menu:\n";
```

Comment:
Again, nothing needs to be changed here, just think about this: how would you decouple the following things from your functions/class - displaying the menu(UI handling), input handling and your actual class logic.